### ISSUE TITLE
Conversation Speed 

### ISSUE THREAD

**Issue by torai** [dularion/streama#182]
hello , well is there any command or some trick to fix the xSpeed ? 

cuz it's started fast around (40 ) then decrease down to 5 or 6 

so is there any help to fix or set a higher number 

thanks

**Comment by srwareham** [comment]
Hi, I'm not a developer on this project, but I think I can confidently tell you what you are looking for is not going to be possible any time in the near future. 

From the little I have played around, streama seems to use the .webm container as an encoding target. This is done so that your browser can playback videos using the html5 video standard. There are a few choices for which container / codec formats are supported by the html5 implementation for most major web browsers but many of these are non-free. I have had the fortune of not needing to muck around in the legalities of such licenses, but it is quite possible that mp4, for example, cannot be supported by streama without acquiring an expensive license--certainly an unlikely eventuality for a (presumably) low-to-no budget open source project. It is also possible the developer, as many do, has a strong preference for "free" software and will not use non-free software when a free analog exists.

The reason this is relevant is because the webm container has very specific requirements for which codecs it supports. Specifically, it supports vorbis and opus for audio, and vp8 and vp9 for video. The less than ideal performance you are seeing is because the current implementations for encoding to vp8 and vp9 are not very fast. The only way for your conversion speed to increase would be for the core developers of the vp8 and vp9 codecs to devise a new implementation that is much more efficient (or for hardware makers to create hardware-accelerated implementations). Alternatively, the developer for this project could add support for other containers / codecs, but as mentioned above this is likely not possible.

Speculation: The original conversion speed you are seeing is likely an artifact of implementation details (by the encoding software used, not streama). That is to say that you are never actually processing frames at that speed, it likely is an inaccurate representation of encoding speed caused by funky measurements in the beginning, or by counting headers and low-overhead muxing as "frames per second" somehow.

TL;DR: Restrictive licenses for standard containers necessitate the use of open alternatives. These open alternatives have not caught up in terms of encoding speed.

**Comment by dularion** [comment]
👍
